raiden/network/proxies/token_network_registry.py

```diff
@@ -76,7 +76,7 @@ def __init__(
         self.node_address = self.client.address
 
     def get_token_network(
-        self, token_address: TokenAddress, block_identifier: BlockSpecification = "latest"
+        self, token_address: TokenAddress, block_identifier: BlockSpecification
     ) -> Optional[TokenNetworkAddress]:
```

Comment:
I did a cursory grep on the code and it seems that every time this function is being called there is a following check for `None` and raising an error. If that is the case, does it make sense to keep the "Optional" return value here? Is there any point of the code where it will be okay to have the token network?

Reply:
Hmm I don't think this makes sense.

 This is a contract proxy. A `None` value simply tells us that the contract has no registered token network. And by grepping you will see that we check in different places how/why that happened and provide an equivalent error message.

Reply:
Right. It just seems that if we are always checking for `None` and throwing an error in case it is None, perhaps the function could throw some exception and the caller of `get_token_network` could handle the exception and provide its own message. Anyway, just a style preference.

Reply:
yeah I understood what you mean, but we try to avoid exceptions as control flow. We use them instead as a way to report errors from the bottom of a call stack and decypher them at the top where they are caught.